From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Mullin <tim@cpanel.net>
Date: Fri, 11 Feb 2022 02:44:46 +0000
Subject: [PATCH 5/5] Check for NULL mutex in apr_global_mutex_child_init

Passing a NULL mutex into apr_global_mutex_child_init
will cause a segmentation fault.
---
 locks/unix/global_mutex.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/locks/unix/global_mutex.c b/locks/unix/global_mutex.c
index 0217351..27eb6df 100644
--- a/locks/unix/global_mutex.c
+++ b/locks/unix/global_mutex.c
@@ -86,6 +86,10 @@ APR_DECLARE(apr_status_t) apr_global_mutex_child_init(
 {
     apr_status_t rv;
 
+    if (*mutex == NULL) {
+        return APR_ENOLOCK;
+    }
+
     rv = apr_proc_mutex_child_init(&((*mutex)->proc_mutex), fname, pool);
     return rv;
 }
